Samples/Windowing/cs-wpf/wpf_packaged_app/TitleBarPage.xaml.cs

```diff
@@ -82,34 +87,80 @@ private void TitlebarCustomBtn_Click(object sender, RoutedEventArgs e)
+                // Bring back the default titlebar
+                _mainWindow.MyTitleBar.Visibility = Visibility.Collapsed;
+
+                // Disable title bar height toggle buttons
+                this.StandardHeightBtn.IsEnabled = false;
+                this.TallHeightBtn.IsEnabled = false;
+
+                // reset the title bar to default state
+                _mainAppWindow.TitleBar.ResetToDefault();
+            }
+        }
+
+        private void SetDragRegionForCustomTitleBar(AppWindow appWindow)
+        {
+            //Infer titlebar height
+            int titleBarHeight = appWindow.TitleBar.Height;
+            _mainWindow.MyTitleBar.Height = titleBarHeight;
+
+            // Get caption button occlusion information
+            // Use LeftInset if you've explicitly set your window layout to RTL or if app language is a RTL language
+            int CaptionButtonOcclusionWidth = appWindow.TitleBar.RightInset;
 
-                // Get caption button occlusion information
-                // Use LeftInset if you've explicitly set your window layout to RTL or if app language is a RTL language
-                int CaptionButtonOcclusionWidth = _mainAppWindow.TitleBar.RightInset;
+            // Define your drag Regions
+            int windowIconWidthAndPadding = (int)_mainWindow.MyWindowIcon.ActualWidth + (int)_mainWindow.MyWindowIcon.Margin.Right;
+            int dragRegionWidth = appWindow.Size.Width - (CaptionButtonOcclusionWidth + windowIconWidthAndPadding);
 
-                // Define your drag Regions
-                int windowIconWidthAndPadding = (int)_mainWindow.MyWindowIcon.ActualWidth + (int)_mainWindow.MyWindowIcon.Margin.Right;
-                int dragRegionWidth = _mainAppWindow.Size.Width - (CaptionButtonOcclusionWidth + windowIconWidthAndPadding);
+            Windows.Graphics.RectInt32[] dragRects = new Windows.Graphics.RectInt32[] { };
+            Windows.Graphics.RectInt32 dragRect;
 
-                Windows.Graphics.RectInt32[] dragRects = new Windows.Graphics.RectInt32[] { };
-                Windows.Graphics.RectInt32 dragRect;
+            dragRect.X = windowIconWidthAndPadding;
+            dragRect.Y = 0;
+            dragRect.Height = titleBarHeight;
+            dragRect.Width = dragRegionWidth;
 
-                dragRect.X = windowIconWidthAndPadding;
-                dragRect.Y = 0;
-                dragRect.Height = titleBarHeight;
-                dragRect.Width = dragRegionWidth;
+            appWindow.TitleBar.SetDragRectangles(dragRects.Append(dragRect).ToArray());
+        }
+
+        private void StandardHeightBtn_Click(object sender, RoutedEventArgs e)
+        {
+            _isTallTitleBar = false;
+            toggleTitlebarHeightOption();
+        }
+        private void TallHeightBtn_Click(object sender, RoutedEventArgs e)
+        {
+            _isTallTitleBar = true;
+            toggleTitlebarHeightOption();
+        }
+        private void toggleTitlebarHeightOption()
```

Comment:
Consider folding these three separate methods in just the one "toggleTitlebarHeightOption" since the separate button click handlers only call that method anyway. This will also fix the problem that _isTallTitleBar value might be out of sync if IsCustomizationSupported returns false or ExtendsIntoTitleBar not having been set/unset.

Reply:
@mevey thoughts?